<?xml version="1.0"?>
<launch>
  <arg name="bagfile" default=""/>

  <!-- KITTI Helper, hack to fake KITTI is a rosbag dataset -->
  <node pkg="kiss_icp" type="kitti_helper_node" name="kitti_helper_node" output="screen"/>
  <node pkg="rosbag" type="play" name="player" args="$(arg bagfile)" output="screen"/>
  <node pkg="rviz" type="rviz" name="rviz" args="-d $(find kiss_icp)/src/cpp/kiss_icp_ros/ros1/rviz/kiss_icp.rviz"/>

  <!-- Odometry node -->
  <node pkg="kiss_icp" type="odometry_node" name="odometry_node" output="screen">
    <param name="odom_frame" value="odom"/>
    <param name="pointcloud_frame" value="pointcloud_frame"/>
    <param name="deskew" value="true"/>
  </node>
</launch>
